models/product-projection/src/transformers.ts

```diff
@@ -79,35 +79,69 @@ const transformers = {
         const metaDescriptionAllLocales = LocalizedString.toLocalizedField(
           fields.metaDescription
         );
-        const productTypeRef = Reference.random()
-          .id(fields.productType.id)
-          .typeId('product-type')
-          .buildGraphql<TReferenceGraphql>();
-        const state = fields.state
-          ? State.random().id(fields.productType.id).buildGraphql<TState>()
-          : undefined;
-        const stateRef = fields.state
-          ? Reference.random()
-              .id(fields.productType.id)
-              .typeId('state')
-              .buildGraphql<TReferenceGraphql>()
-          : undefined;
-        const taxCategory = fields.taxCategory
-          ? TaxCategory.random()
-              .id(fields.productType.id)
-              .buildGraphql<TTaxCategory>()
-          : undefined;
-        const taxCategoryRef = fields.taxCategory
-          ? Reference.random()
-              .id(fields.productType.id)
-              .typeId('tax-category')
-              .buildGraphql<TReferenceGraphql>()
-          : undefined;
-        const categoriesRef = fields.categories.map((category) => ({
-          id: category.id,
-          typeId: 'category' as const,
-          __typename: 'Reference' as const,
-        }));
+
+        const restProductType = buildField(fields.productType, 'rest');
+        const productTypeRef = buildField(
+          fields.productType,
+          'graphql'
+        ) as TReferenceGraphql;
+        const productType = ProductType.random()
+          .description(restProductType.obj?.description || '')
+          .id(restProductType.id)
+          .key(restProductType.obj?.key || '')
+          .name(restProductType.obj?.name || '')
+          .buildGraphql<TProductTypeGraphql>();
+
+        let state: TStateGraphql | null = null;
+        let stateRef: TReferenceGraphql | null = null;
+        if (fields.state) {
+          const restState = buildField(fields.state, 'rest');
+          stateRef = buildField(fields.state, 'graphql') as TReferenceGraphql;
```

Comment:
Since the nested model in the default representation holds a `Reference`, it would be great that we could build the `obj` property of the reference for the graphql output but we can't build nested properties in isolation and the graphql transformer of the `Reference` model removes the `obj` property since it's not part of that representation. So the alternative I found is to build the rest representation and use some its values to create a new graphql model and populate it with it.

Technically this is not 100% right as we would need to overwrite all create graphql instance with the values from the rest one but I can't see a way to do it in a maintainable way.